src/test/java/picard/sam/testers/SamFileTester.java

```diff
@@ -364,10 +374,37 @@ public void runTest() {
         }
     }
 
-    private File createInputFile() {
+    private File createInputFile(final String extension) throws IOException {
         // Create the input file
-        final File input = new File(outputDir, "input.sam");
-        final SAMFileWriter writer = new SAMFileWriterFactory().makeSAMOrBAMWriter(samRecordSetBuilder.getHeader(), true, input);
+        final File input = new File(outputDir, "input" + extension);
+
+        final SAMFileWriterFactory samFileWriterFactory = new SAMFileWriterFactory();
+
+        final SAMFileWriter writer;
+        if (extension.equals(".cram")) {
+            final Path fasta = fastaFiles.computeIfAbsent(samRecordSetBuilder.getHeader(), h -> {
+
+                final Path fastaDir = IOUtil.createTempDir("SamFileTester", "").toPath();
+                IOUtil.deleteOnExit(fastaDir);
+                final Path newFasta = fastaDir.resolve("input.fasta");
+                IOUtil.deleteOnExit(newFasta);
+                IOUtil.deleteOnExit(ReferenceSequenceFileFactory.getFastaIndexFileName(newFasta));
+                IOUtil.deleteOnExit(ReferenceSequenceFileFactory.getDefaultDictionaryForReferenceSequence(newFasta));
+
+                try {
+                    samRecordSetBuilder.writeRandomReference(newFasta);
```

Comment:
So this would appear to write out a random reference of 200,000,000 bases per each contig (of which there are the customary 23 by default here). That seems like a lot to write out for every single cram test individually. Perhaps this should be cached between cram tests on the same test invocation or you should explicitly set the default base length to something smaller to save on runtime spent making dummy fasta files.

Reply:
only for crams...which is why modifyTesterForCrams cuts the size of the reference down significantly...I'll add an error if more than 10^7 bases are about to be written...